api/internal/localizer/util.go

```diff
@@ -132,25 +135,65 @@ func locFilePath(fileURL string) string {
 	// File urls must have http or https scheme, so it is safe to use url.Parse.
 	u, err := url.Parse(fileURL)
 	if err != nil {
-		log.Fatalf("cannot parse validated file url %q: %s", fileURL, err.Error())
+		log.Panicf("cannot parse validated file url %q: %s", fileURL, err)
```

Comment:
Why do you want to panic? A user in general should never run into a panic, it means something has gone terribly and unexpectedly wrong.

Reply:
I originally chose to `log.Fatal` because `fileLoader` `url.Parse`s file urls to determine if they have the `https`, `http` scheme. In this case, then, if `url.Parse` returns an error, something has gone unexpectedly wrong. I don't  consider this a tight coupling because if `fileLoader` no longer checks for `https` scheme with `url.Parse`, we may have to change other aspects of the localized file path in `locFilePath`.

I haven't changed the logic in this line. The change only reflects Katrina's recommendation to replace `log.Fatal` with `log.Panic`: https://github.com/kubernetes-sigs/kustomize/pull/4867#discussion_r1028622515

Reply:
thanks for the explanation, happy to learn that we decide to use `log.Panicf`.